sql/src/main/java/io/crate/expression/tablefunctions/PgGetKeywordsFunction.java

```diff
@@ -0,0 +1,137 @@
+    private static final RelationName REL_NAME = new RelationName(PgCatalogSchemaInfo.NAME, NAME);
+    private static final PgGetKeywordsFunction INSTANCE = new PgGetKeywordsFunction();
+    private final FunctionInfo info;
+
+    public static void register(TableFunctionModule module) {
+        module.register(
+            FUNCTION_NAME,
+            new BaseFunctionResolver(FuncParams.NONE) {
+
+                @Override
+                public FunctionImplementation getForTypes(List<DataType> argTypes) throws IllegalArgumentException {
+                    assert argTypes.isEmpty() : "argument types for pg_get_keywords must be empty due to FuncParams definition";
+                    return PgGetKeywordsFunction.INSTANCE;
+                }
+            }
+        );
+    }
+
+    static class Columns {
+        static final ColumnIdent WORD = new ColumnIdent("word");
+        static final ColumnIdent CATCODE = new ColumnIdent("catcode");
+        static final ColumnIdent CATDESC = new ColumnIdent("catdesc");
+    }
+
+    public PgGetKeywordsFunction() {
+        info = new FunctionInfo(
+            new FunctionIdent(FUNCTION_NAME, List.of()),
+            ObjectType.untyped(),
+            FunctionInfo.Type.TABLE
+        );
+    }
+
+    @Override
+    public TableInfo createTableInfo() {
+        LinkedHashMap<ColumnIdent, Reference> columnMap = new LinkedHashMap<>();
+        columnMap.put(
+            Columns.WORD,
+            new Reference(new ReferenceIdent(REL_NAME, Columns.WORD), RowGranularity.DOC, DataTypes.STRING, 0, null)
+        );
+        columnMap.put(
+            Columns.CATCODE,
+            new Reference(new ReferenceIdent(REL_NAME, Columns.CATCODE), RowGranularity.DOC, DataTypes.STRING, 1, null)
+        );
+        columnMap.put(
+            Columns.CATDESC,
+            new Reference(new ReferenceIdent(REL_NAME, Columns.CATDESC), RowGranularity.DOC, DataTypes.STRING, 2, null)
+        );
+        return new StaticTableInfo<>(REL_NAME, columnMap, columnMap.values(), List.of()) {
+
+            @Override
+            public Routing getRouting(ClusterState state,
+                                      RoutingProvider routingProvider,
+                                      WhereClause whereClause,
+                                      RoutingProvider.ShardSelection shardSelection,
+                                      SessionContext sessionContext) {
+                return Routing.forTableOnSingleNode(REL_NAME, state.getNodes().getLocalNodeId());
+            }
+        };
+    }
+
+    @Override
+    public Iterable<Row> evaluate(TransactionContext txnCtx, Input<List<Object>>... args) {
+        return () -> Identifiers.KEYWORDS.stream()
+            .map(keyword -> (Row) new RowN(
+                keyword.getWord().toLowerCase(Locale.ENGLISH),
+                keyword.isReserved() ? "R" : "U",
+                keyword.isReserved() ? "reserved" : "unreserved"
+            ))
```

Comment:
wouldn't it make sense to store this map result into a static var?

Reply:
I was considering it, but felt that the function is likely not used often and the performance overhead is small enough. So it does not justify having another List of rows as permanent memory footprint.

I was also wondering if I should add `categoryCode()` and `categoryDescription` getters to the `Keyword` class. But the output felt to be specific to `pg_get_keywords`, so I put it here.

But I could change it to use a shared row object. What do you think?

Reply:
I see, shared row sounds good.